src/routes/docs/tutorials/apple/step-5/+page.markdoc

```diff
@@ -0,0 +1,402 @@
+        } else {
+            routes[routes.count - 1] = screen
+        }
+    }
+    
+    func pop() {
+        routes.removeLast()
+    }
+    
+    func popUntil(predicate: (Route) -> Bool) {
+        if let last = routes.popLast() {
+            guard predicate(last) else {
+                popUntil(predicate: predicate)
+                return
+            }
+        }
+    }
+    
+    func reset() {
+        routes = []
+    }
+}
+```
+
+# ContentView {% #contentview %}
+
+Update the ContentView.swift to reflect some changes related to our navigation
+
+```swift
+import SwiftUI
+
+struct ContentView: View {
+    
+    @EnvironmentObject private var router: Router
+    @EnvironmentObject private var userViewModel: UserViewModel
+    @EnvironmentObject private var snackBarService: SnackBarService
+    @Environment(\.colorScheme) private var theme
+    
+    
+    var body: some View {
+        NavigationStack(path: $router.routes) {
+            VStack {
+                
+                Image(theme == .dark ? "appwrite_light" : "appwrite_dark")
+                    .frame(width: 132, height: 24)
+                
+            }
+            .task {
+                
+                let isAuthenticated = await userViewModel.getCurrentSession()
+                
+                if !isAuthenticated {
+                    router.pushReplacement(.auth)
+                } else {
+                    router.pushReplacement(.home)
+                }
+            }
+            .navigationDestination(for: Route.self, destination: { $0 })
+            
+        }
+        .overlay(alignment: .top) {
+            if (snackBarService.snackBarState?.hasError == true) {
+                SnackbarView(text: snackBarService.snackBarState?.error.localizedDescription ?? "An error occured")
+            }
+        }
+        
+    }
+}
+```
+
+Note: You can get the Appwrite logo on Appwrite website and add it to your project assets
+
+# Inject services and view models {% #inject-services-and-view-models %}
+
+Let's inject Appwrite Service and Snarbar Service into our View Model then inject to our app using Environment Object. Update the entry file with the code below.
+
+```swift
+import SwiftUI
+
+@main
+struct appwrite_hacktoberfestApp: App {
+    
+    private var appwriteService = AppwriteService()
+    private var snackbarService = SnackBarService()
+    
+    var body: some Scene {
+        WindowGroup {
+            ContentView()
+                .environmentObject(Router())
+                .environmentObject(UserViewModel(appwriteService, snackbarService))
+                .environmentObject(HabitViewModel(appwriteService, snackbarService))
+                .environmentObject(snackbarService)
+        }
+    }
+}
+```
+
+Note: Change **appwrite_hacktoberfestApp** to your project name
+
+
+# User view model {% #user-view-models %}
+
+Add the following code to UserViewModel.swift inside **Home/ViewModels** folder.
```

Comment:
Please use code tags for files and paths.

Reply:
Also Home/ViewModels isn't the right path, probably. The name depends on name of app used.